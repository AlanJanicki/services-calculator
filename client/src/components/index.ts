export { Card } from './Card/Card';
export { Checkbox } from './Checkbox/Checkbox';
export { Select } from './Select/Select';
export { Spinner } from './Spinner/Spinner';
