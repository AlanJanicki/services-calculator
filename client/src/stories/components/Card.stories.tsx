import { Card } from '../../components';

export default {
  component: Card,
  tags: ['autodocs'],
  title: 'Card'
};

export const Default = {
  args: {
    header: 'Kalkulator us≈Çug'
  }
};
