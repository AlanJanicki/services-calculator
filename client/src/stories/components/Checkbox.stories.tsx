import { Checkbox } from '../../components';

export default {
  component: Checkbox,
  tags: ['autodocs'],
  title: 'Checkbox'
};

export const Default = {
  args: {
    name: 'Internet'
  }
};
