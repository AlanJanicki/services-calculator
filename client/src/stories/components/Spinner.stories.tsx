import { Spinner } from '../../components';

export default {
  component: Spinner,
  tags: ['autodocs'],
  title: 'Spinner'
};

export const Default = {};
