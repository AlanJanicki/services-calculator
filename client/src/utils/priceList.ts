import { Service } from '../models';

export const getServicePrice = ({ fixedPrice, prices }: Service, yearIndex: number) =>
  fixedPrice ?? (prices ?? [])[yearIndex];

export const getTotal = (services: Service[], yearIndex: number) =>
  services.reduce((acc, next) => acc + getServicePrice(next, yearIndex), 0);

export const getInvalidServiceInfo = (missingComplementaryServices: Service[]) =>
  `Konieczne jest jeszcze zakupienie usÅ‚ug: ${missingComplementaryServices
    .map(({ name }) => name)
    .join(', ')}`;
