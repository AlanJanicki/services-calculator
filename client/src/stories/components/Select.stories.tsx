import { Select } from '../../components';
import priceList from '../../mocks/servicesPriceList.json';

export default {
  component: Select,
  tags: ['autodocs'],
  title: 'Select'
};

export const Default = {
  args: {
    label: 'Wybierz rok',
    options: priceList.years
  }
};
