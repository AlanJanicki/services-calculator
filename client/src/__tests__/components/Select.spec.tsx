import React from 'react';

import { Select } from '../../components';
import priceList from '../../mocks/servicesPriceList.json';
import { renderWithSnapshot } from '../common';

describe('<Select/>', () => {
  it('matches snapshot', () =>
    renderWithSnapshot(
      <Select id='years' label='Wybierz rok' options={priceList.years} onChange={jest.fn} />
    ));
});
