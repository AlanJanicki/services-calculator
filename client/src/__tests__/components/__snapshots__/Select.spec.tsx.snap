// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select/> matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
}

.c0 label {
  margin: 0 auto 1rem auto;
}

.c0 select {
  cursor: pointer;
  padding: 0.1rem 0.5rem;
  border: 1px solid #dfdfdf;
  border-radius: .25rem;
}

@media (min-width:992px) {
  .c0 label,
  .c0 select {
    font-size: 1.2rem;
  }
}

<div
    class="c0"
  >
    <label
      for="years"
    >
      Wybierz rok
    </label>
    <select
      id="years"
    >
      <option
        value="2023"
      >
        2023
      </option>
      <option
        value="2024"
      >
        2024
      </option>
      <option
        value="2025"
      >
        2025
      </option>
    </select>
  </div>
</DocumentFragment>
`;
